File: kotlin-ide/j2k/new/testData/newJ2k/function/referenceToConstructor.java
// RUNTIME_WITH_FULL_JDK

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class TestLambda {
    public static void main(String[] args) {
        List<String> names = Arrays.asList("A", "B");
        List<Person> people = names.stream().map(Person::new).collect(Collectors.toList());
    }
}

File: nj2k/testData/newJ2k/function/referenceToConstructor.kt
import java.util.Arrays
import java.util.stream.Collectors

object TestLambda {
    @JvmStatic
    fun main(args: Array<String>) {
        val names = Arrays.asList("A", "B")
        val people = names.stream().map { name: String? -> Person(name) }.collect(Collectors.toList())
    }
}